### ISSUE TITLE
void-variable consult--regexp-compiler recently introduced error?

### ISSUE THREAD

**Issue by jsilve24** [minad/consult#394]
First off, thank you for putting together such a nice package. 

I am using doom and recently ran `doom sync` and ever since I am having trouble with vertico/consult. 

I am getting  
`Error in post-command-hook (vertico--exhibit): (void-variable consult--regexp-compiler)` 
and 
`Failed to load deferred package orderless: (void-variable consult--regexp-compiler)`

I am not sure how to debug further other than recognizing that a number of the recent commits in this repo seem to mention this variable (`consult--regex-compiler`) could this be a recently intorduced bug?

**Comment by minad** [comment]
As far as I know consult--regexp-compiler is not yet used by the doom vertico module. Which command makes problems? Cc @iyefrat

**Comment by jsilve24** [comment]
There are quite a number that work and a number that don't, I have not yet figgured out the pattern but I am guessing its which ones depend on consult. 

Dont work:
For these commands, as soon as I try to type anything in I get that error. 
- consult-recent-file
- +vertico/switch-workspace-buffer
- projectile-switch-project

Does work:
- find-file

**Comment by iyefrat** [comment]
Yeah the version of consult on the latest `develop` is pinned to a commit before this feature was introduced. Do you have `(unpin! consult)` in your `packages.el`? If so, this is inadvisable, the packages in this ecosystem are a bit young, and still sometimes go through interface changes in between stable versions, but if you stick to the pinned versions in doom that shouldn't give you any trouble.

Also, if you encounter a bug in doom, your best bet is asking for help on the discord server or opening an issue on doom's issue tracker if you can't reproduce it in `emacs -Q`, since it's more likely to be a doom thing than an upstream bug in that case.

**Comment by jsilve24** [comment]
| Do you have (unpin! consult) in your packages.el?

Not that I know of...

let me try emacs -Q (going to have to look up what that is)

**Comment by iyefrat** [comment]
This is what `emacs -Q` is supposed to do, you start up vanilla emacs, install only the packages involved, and try and reproduce the bug. That way you know that `consult` is actually causing the bug and not something else. In this case it's likely you somehow managed to mix and match consult versions, so I doubt that the bug would be reproducible there. Try to run the following:
```
doom upgrade && doom clean && doom purge && doom build -r
```
That should clean out any unintentional cruft in your doom install that might cause this.

(@minad feel free to close this thread and tell us to go somewhere else if you want)

**Comment by jsilve24** [comment]
Thanks @iyefrat, I just ran all this and no luck. Same problem.

**Comment by iyefrat** [comment]
Try `rm -rf ~/.emacs.d/.local/straight/repos/consult && doom sync`.

**Comment by jsilve24** [comment]
Man this sucks -- still having the same problem.

**Comment by iyefrat** [comment]
can you post your private config?

**Comment by jsilve24** [comment]
Yes absolutely thank you! give me one second

**Comment by jsilve24** [comment]
Just made this public. 
https://github.com/jsilve24/dotfiles/tree/4cb975df51da53d9108678a19c675b72792d29c8/home/.doom.d

**Comment by minad** [comment]
The issue is properly affe:

https://github.com/jsilve24/dotfiles/blob/4cb975df51da53d9108678a19c675b72792d29c8/home/.doom.d/config.el#L107-L112

1. affe should not be loaded eagerly
2. you should pin affe to a version compatible with the doom consult version

**Comment by minad** [comment]
@jsilve24 it should suffice to pin affe to https://github.com/minad/affe/commit/0e224a274471d27e986ce650dbb8e9252c2b7b2a
